fsspec/implementations/arrow.py

```diff
@@ -155,9 +158,10 @@ def _open(self, path, mode="rb", block_size=None, **kwargs):
             raise ValueError(f"unsupported mode for Arrow filesystem: {mode!r}")
 
         _kwargs = {}
-        if PYARROW_VERSION[0] >= 4:
-            # disable compression auto-detection
-            _kwargs["compression"] = None
+        if mode != "rb" or not seekable:
+            if PYARROW_VERSION[0] >= 4:
+                # disable compression auto-detection
+                _kwargs["compression"] = None
```

Comment:
I guess we don't test this, since only one version of pyarrow is installed? I don't see what the practical consequences will be.

Reply:
It was introduced in #911 to fix issue #904.

As we state `"pyarrow >= 1"` in setup.py, if we remove the check here would be a backwards compatibility issue, I guess.